tests/test_category.py

```diff
@@ -0,0 +1,116 @@
+import json
+
+from django.contrib.auth.models import User
+from rest_framework.test import APITestCase
+from django.db.models import Q
+from django.core.exceptions import ValidationError
+
+from . import TomatoTest
+from wallet.serializers import CategorySerializer
+from wallet.views import CategoryViewSet
+from wallet.models import Category
+
+
+class CategoryTest(TomatoTest):
+    def setUp(self):
+        super().setUp()
+
+    def test_list_anonymousUser(self):
+        response = self.client.get('/category/')
+        self.assertEqual(response.status_code, 403)
+
+    def test_list(self):
+        # tes_create_andt category list shows only categroies with no owner or
+        # owner is current user.
+        Category.objects.create(name='default')
+        Category.objects.create(name='sbs category', owner=self.sb)
+        Category.objects.create(name='tomatos category', owner=self.user)
+
+        self.assertTrue(self.client.login(**self.user_info))
+        response = self.client.get('/category/')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content)['count'], 2)
+
+        serializer = CategorySerializer(
```

Comment:
If you just `assert response.content['results'] == [{"id": 1, ...}, {"id": 2, ...}...]`, then your test will still work if you change how you implement the API. This is a great benefit when it comes to refactor your code, for example, if you decide to use some other rest framework which doesn't have Serializers.

Reply:
The possibility to use another framework is pretty small. The possibility to change the response structure is more possible. Once you changed the structure, you have to update all of your tests. Make sure the response structure correctly is serializer's work. As you said you don't have to check a package's work, you only have to make sure you are using it correctly. So check the response once is fare enough, Because create, edit, get one API is returning the same fields. Check out [this](https://realpython.com/test-driven-development-of-a-django-restful-api/) article.